### PR TITLE
Pin dependencies

### PR DESCRIPTION
**Pull request by krypton-build-user**
This PR contains the following updates:

| Package | Type | Update | Change |
|---|---|---|---|
| [@aws-crypto/sha256-js](https://togithub.com/aws/aws-sdk-js-crypto-helpers/tree/master/packages/sha256-js) ([source](https://togithub.com/aws/aws-sdk-js-crypto-helpers)) | dependencies | pin | [`^2.0.2` -> `2.0.2`](https://renovatebot.com/diffs/npm/@aws-crypto%2fsha256-js/2.0.2/2.0.2) |
| [@aws-sdk/client-sts](https://togithub.com/aws/aws-sdk-js-v3/tree/main/clients/client-sts) ([source](https://togithub.com/aws/aws-sdk-js-v3)) | dependencies | pin | [`^3.245.0` -> `3.245.0`](https://renovatebot.com/diffs/npm/@aws-sdk%2fclient-sts/3.245.0/3.245.0) |
| [@aws-sdk/signature-v4](https://togithub.com/aws/aws-sdk-js-v3/tree/main/packages/signature-v4) ([source](https://togithub.com/aws/aws-sdk-js-v3)) | dependencies | pin | [`^3.226.0` -> `3.226.0`](https://renovatebot.com/diffs/npm/@aws-sdk%2fsignature-v4/3.226.0/3.226.0) |
| [@brightcove/kacl](https://togithub.com/brightcove/kacl) | devDependencies | pin | [`^0.1.11` -> `0.1.11`](https://renovatebot.com/diffs/npm/@brightcove%2fkacl/0.1.11/0.1.11) |
| [@getify/eslint-plugin-proper-arrows](https://togithub.com/getify/eslint-plugin-proper-arrows) | devDependencies | pin | [`^11.0.3` -> `11.0.3`](https://renovatebot.com/diffs/npm/@getify%2feslint-plugin-proper-arrows/11.0.3/11.0.3) |
| [@julian_cataldo/remark-lint-frontmatter-schema](https://togithub.com/JulianCataldo/remark-lint-frontmatter-schema) | devDependencies | pin | [`^3.7.6` -> `3.7.6`](https://renovatebot.com/diffs/npm/@julian_cataldo%2fremark-lint-frontmatter-schema/3.7.6/3.7.6) |
| [@release-it/keep-a-changelog](https://togithub.com/release-it/keep-a-changelog) | devDependencies | pin | [`^3.1.0` -> `3.1.0`](https://renovatebot.com/diffs/npm/@release-it%2fkeep-a-changelog/3.1.0/3.1.0) |
| [@types/aws-sdk](https://togithub.com/aws/aws-sdk-js) | devDependencies | pin | [`^2.7.0` -> `2.7.0`](https://renovatebot.com/diffs/npm/@types%2faws-sdk/2.7.0/2.7.0) |
| [@types/browserify](https://togithub.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/browserify) ([source](https://togithub.com/DefinitelyTyped/DefinitelyTyped)) | devDependencies | pin | [`^12.0.37` -> `12.0.37`](https://renovatebot.com/diffs/npm/@types%2fbrowserify/12.0.37/12.0.37) |
| [@types/chai](https://togithub.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/chai) ([source](https://togithub.com/DefinitelyTyped/DefinitelyTyped)) | devDependencies | pin | [`^4.3.4` -> `4.3.4`](https://renovatebot.com/diffs/npm/@types%2fchai/4.3.4/4.3.4) |
| [@types/clone](https://togithub.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/clone) ([source](https://togithub.com/DefinitelyTyped/DefinitelyTyped)) | devDependencies | pin | [`^2.1.1` -> `2.1.1`](https://renovatebot.com/diffs/npm/@types%2fclone/2.1.1/2.1.1) |
| [@types/exorcist](https://togithub.com/DefinitelyTyped/DefinitelyTyped) | devDependencies | pin | [`^2.0.0` -> `2.0.0`](https://renovatebot.com/diffs/npm/@types%2fexorcist/2.0.0/2.0.0) |
| [@types/express](https://togithub.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/express) ([source](https://togithub.com/DefinitelyTyped/DefinitelyTyped)) | devDependencies | pin | [`^4.17.15` -> `4.17.15`](https://renovatebot.com/diffs/npm/@types%2fexpress/4.17.15/4.17.15) |
| [@types/fs-extra](https://togithub.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/fs-extra) ([source](https://togithub.com/DefinitelyTyped/DefinitelyTyped)) | devDependencies | pin | [`^11.0.1` -> `11.0.1`](https://renovatebot.com/diffs/npm/@types%2ffs-extra/11.0.1/11.0.1) |
| [@types/glob](https://togithub.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/glob) ([source](https://togithub.com/DefinitelyTyped/DefinitelyTyped)) | devDependencies | pin | [`^8.0.0` -> `8.0.0`](https://renovatebot.com/diffs/npm/@types%2fglob/8.0.0/8.0.0) |
| [@types/js-yaml](https://togithub.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/js-yaml) ([source](https://togithub.com/DefinitelyTyped/DefinitelyTyped)) | devDependencies | pin | [`^4.0.5` -> `4.0.5`](https://renovatebot.com/diffs/npm/@types%2fjs-yaml/4.0.5/4.0.5) |
| [@types/mime](https://togithub.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/mime) ([source](https://togithub.com/DefinitelyTyped/DefinitelyTyped)) | devDependencies | pin | [`^3.0.1` -> `3.0.1`](https://renovatebot.com/diffs/npm/@types%2fmime/3.0.1/3.0.1) |
| [@types/mocha](https://togithub.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/mocha) ([source](https://togithub.com/DefinitelyTyped/DefinitelyTyped)) | devDependencies | pin | [`^10.0.1` -> `10.0.1`](https://renovatebot.com/diffs/npm/@types%2fmocha/10.0.1/10.0.1) |
| [@types/mock-fs](https://togithub.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/mock-fs) ([source](https://togithub.com/DefinitelyTyped/DefinitelyTyped)) | devDependencies | pin | [`^4.13.1` -> `4.13.1`](https://renovatebot.com/diffs/npm/@types%2fmock-fs/4.13.1/4.13.1) |
| [@types/node](https://togithub.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/node) ([source](https://togithub.com/DefinitelyTyped/DefinitelyTyped)) | devDependencies | pin | [`^18.11.18` -> `18.11.18`](https://renovatebot.com/diffs/npm/@types%2fnode/18.11.18/18.11.18) |
| [@types/node-fetch](https://togithub.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/node-fetch) ([source](https://togithub.com/DefinitelyTyped/DefinitelyTyped)) | devDependencies | pin | [`^2.6.1` -> `2.6.1`](https://renovatebot.com/diffs/npm/@types%2fnode-fetch/2.6.1/2.6.1) |
| [@types/pascalcase](https://togithub.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/pascalcase) ([source](https://togithub.com/DefinitelyTyped/DefinitelyTyped)) | devDependencies | pin | [`^1.0.1` -> `1.0.1`](https://renovatebot.com/diffs/npm/@types%2fpascalcase/1.0.1/1.0.1) |
| [@types/proxyquire](https://togithub.com/DefinitelyTyped/DefinitelyTyped) | devDependencies | pin | [`^1.3.28` -> `1.3.28`](https://renovatebot.com/diffs/npm/@types%2fproxyquire/1.3.28/1.3.28) |
| [@types/qs](https://togithub.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/qs) ([source](https://togithub.com/DefinitelyTyped/DefinitelyTyped)) | devDependencies | pin | [`^6.9.7` -> `6.9.7`](https://renovatebot.com/diffs/npm/@types%2fqs/6.9.7/6.9.7) |
| [@types/request](https://togithub.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/request) ([source](https://togithub.com/DefinitelyTyped/DefinitelyTyped)) | devDependencies | pin | [`^2.48.8` -> `2.48.8`](https://renovatebot.com/diffs/npm/@types%2frequest/2.48.8/2.48.8) |
| [@types/request-promise-native](https://togithub.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/request-promise-native) ([source](https://togithub.com/DefinitelyTyped/DefinitelyTyped)) | devDependencies | pin | [`^1.0.18` -> `1.0.18`](https://renovatebot.com/diffs/npm/@types%2frequest-promise-native/1.0.18/1.0.18) |
| [@types/semver](https://togithub.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/semver) ([source](https://togithub.com/DefinitelyTyped/DefinitelyTyped)) | devDependencies | pin | [`^7.3.13` -> `7.3.13`](https://renovatebot.com/diffs/npm/@types%2fsemver/7.3.13/7.3.13) |
| [@types/sha.js](https://togithub.com/DefinitelyTyped/DefinitelyTyped) | devDependencies | pin | [`^2.4.0` -> `2.4.0`](https://renovatebot.com/diffs/npm/@types%2fsha.js/2.4.0/2.4.0) |
| [@types/sinon](https://togithub.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/sinon) ([source](https://togithub.com/DefinitelyTyped/DefinitelyTyped)) | devDependencies | pin | [`^10.0.13` -> `10.0.13`](https://renovatebot.com/diffs/npm/@types%2fsinon/10.0.13/10.0.13) |
| [@types/uglify-js](https://togithub.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/uglify-js) ([source](https://togithub.com/DefinitelyTyped/DefinitelyTyped)) | devDependencies | pin | [`^3.17.1` -> `3.17.1`](https://renovatebot.com/diffs/npm/@types%2fuglify-js/3.17.1/3.17.1) |
| [@types/url-parse](https://togithub.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/url-parse) ([source](https://togithub.com/DefinitelyTyped/DefinitelyTyped)) | devDependencies | pin | [`^1.4.8` -> `1.4.8`](https://renovatebot.com/diffs/npm/@types%2furl-parse/1.4.8/1.4.8) |
| [@types/uuid](https://togithub.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/uuid) ([source](https://togithub.com/DefinitelyTyped/DefinitelyTyped)) | devDependencies | pin | [`^9.0.0` -> `9.0.0`](https://renovatebot.com/diffs/npm/@types%2fuuid/9.0.0/9.0.0) |
| [@types/xml2js](https://togithub.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/xml2js) ([source](https://togithub.com/DefinitelyTyped/DefinitelyTyped)) | devDependencies | pin | [`^0.4.11` -> `0.4.11`](https://renovatebot.com/diffs/npm/@types%2fxml2js/0.4.11/0.4.11) |
| [@types/yargs](https://togithub.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/yargs) ([source](https://togithub.com/DefinitelyTyped/DefinitelyTyped)) | devDependencies | pin | [`^17.0.19` -> `17.0.19`](https://renovatebot.com/diffs/npm/@types%2fyargs/17.0.19/17.0.19) |
| [ansi-regex](https://togithub.com/chalk/ansi-regex) | resolutions | pin | [`^5.0.1` -> `5.0.1`](https://renovatebot.com/diffs/npm/ansi-regex/5.0.1/5.0.1) |
| [aws-sdk](https://togithub.com/aws/aws-sdk-js) | devDependencies | pin | [`^2.1290.0` -> `2.1290.0`](https://renovatebot.com/diffs/npm/aws-sdk/2.1290.0/2.1290.0) |
| [browserify](https://togithub.com/browserify/browserify) | dependencies | pin | [`^17.0.0` -> `17.0.0`](https://renovatebot.com/diffs/npm/browserify/17.0.0/17.0.0) |
| [buffer](https://togithub.com/feross/buffer) | devDependencies | pin | [`^6.0.3` -> `6.0.3`](https://renovatebot.com/diffs/npm/buffer/6.0.3/6.0.3) |
| [chai](http://chaijs.com) ([source](https://togithub.com/chaijs/chai)) | devDependencies | pin | [`^4.3.7` -> `4.3.7`](https://renovatebot.com/diffs/npm/chai/4.3.7/4.3.7) |
| [clone](https://togithub.com/pvorb/node-clone) | dependencies | pin | [`^2.1.2` -> `2.1.2`](https://renovatebot.com/diffs/npm/clone/2.1.2/2.1.2) |
| [cross-fetch](https://togithub.com/lquixada/cross-fetch) | dependencies | pin | [`^3.1.5` -> `3.1.5`](https://renovatebot.com/diffs/npm/cross-fetch/3.1.5/3.1.5) |
| [cspell](https://streetsidesoftware.github.io/cspell/) ([source](https://togithub.com/streetsidesoftware/cspell)) | devDependencies | pin | [`^6.19.2` -> `6.19.2`](https://renovatebot.com/diffs/npm/cspell/6.19.2/6.19.2) |
| [dts-bundle-generator](https://togithub.com/timocov/dts-bundle-generator) | devDependencies | pin | [`^6.13.0` -> `6.13.0`](https://renovatebot.com/diffs/npm/dts-bundle-generator/6.13.0/6.13.0) |
| [es6-promise](https://togithub.com/stefanpenner/es6-promise) | dependencies | pin | [`^4.2.8` -> `4.2.8`](https://renovatebot.com/diffs/npm/es6-promise/4.2.8/4.2.8) |
| [esbuild](https://togithub.com/evanw/esbuild) | dependencies | pin | [`^0.16.16` -> `0.16.16`](https://renovatebot.com/diffs/npm/esbuild/0.16.16/0.16.16) |
| [eslint-plugin-ban](https://togithub.com/remithomas/eslint-plugin-ban) | devDependencies | pin | [`^1.6.0` -> `1.6.0`](https://renovatebot.com/diffs/npm/eslint-plugin-ban/1.6.0/1.6.0) |
| [eslint-plugin-filenames](https://togithub.com/selaux/eslint-plugin-filenames) | devDependencies | pin | [`^1.3.2` -> `1.3.2`](https://renovatebot.com/diffs/npm/eslint-plugin-filenames/1.3.2/1.3.2) |
| [eslint-plugin-local](https://togithub.com/taskworld/eslint-plugin-local) | devDependencies | pin | [`^1.0.0` -> `1.0.0`](https://renovatebot.com/diffs/npm/eslint-plugin-local/1.0.0/1.0.0) |
| [eslint-plugin-prefer-let](https://togithub.com/thefrontside/javascript) | devDependencies | pin | [`^3.0.1` -> `3.0.1`](https://renovatebot.com/diffs/npm/eslint-plugin-prefer-let/3.0.1/3.0.1) |
| [exorcist](https://togithub.com/thlorenz/exorcist) | dependencies | pin | [`^2.0.0` -> `2.0.0`](https://renovatebot.com/diffs/npm/exorcist/2.0.0/2.0.0) |
| [fs-extra](https://togithub.com/jprichardson/node-fs-extra) | dependencies | pin | [`^11.1.0` -> `11.1.0`](https://renovatebot.com/diffs/npm/fs-extra/11.1.0/11.1.0) |
| [glob](https://togithub.com/isaacs/node-glob) | devDependencies | pin | [`^8.0.3` -> `8.0.3`](https://renovatebot.com/diffs/npm/glob/8.0.3/8.0.3) |
| [got](https://togithub.com/sindresorhus/got) | resolutions | pin | [`^12.1.0` -> `12.5.3`](https://renovatebot.com/diffs/npm/got/12.5.3/12.5.3) |
| [handlebars](http://www.handlebarsjs.com/) ([source](https://togithub.com/wycats/handlebars.js)) | dependencies | pin | [`^4.7.7` -> `4.7.7`](https://renovatebot.com/diffs/npm/handlebars/4.7.7/4.7.7) |
| [isolated-vm](https://togithub.com/laverdet/isolated-vm) | optionalDependencies | pin | [`=4.4.1 \|\| >4.4.2` -> `4.4.1`](https://renovatebot.com/diffs/npm/isolated-vm/4.4.1/4.4.1) |
| [js-yaml](https://togithub.com/nodeca/js-yaml) | devDependencies | pin | [`^4.1.0` -> `4.1.0`](https://renovatebot.com/diffs/npm/js-yaml/4.1.0/4.1.0) |
| [jsonpath-plus](https://togithub.com/s3u/JSONPath) | dependencies | pin | [`^7.2.0` -> `7.2.0`](https://renovatebot.com/diffs/npm/jsonpath-plus/7.2.0/7.2.0) |
| [jsprim](https://togithub.com/joyent/node-jsprim) | resolutions | pin | [`^2.0.2` -> `2.0.2`](https://renovatebot.com/diffs/npm/jsprim/2.0.2/2.0.2) |
| [marked](https://marked.js.org) ([source](https://togithub.com/markedjs/marked)) | resolutions | pin | [`^4.0.10` -> `4.2.12`](https://renovatebot.com/diffs/npm/marked/4.2.12/4.2.12) |
| [mersenne-twister](https://togithub.com/boo1ean/mersenne-twister) | dependencies | pin | [`^1.1.0` -> `1.1.0`](https://renovatebot.com/diffs/npm/mersenne-twister/1.1.0/1.1.0) |
| [mime](https://togithub.com/broofa/mime) | devDependencies | pin | [`^3.0.0` -> `3.0.0`](https://renovatebot.com/diffs/npm/mime/3.0.0/3.0.0) |
| [minimist](https://togithub.com/minimistjs/minimist) | resolutions | pin | [`^1.2.6` -> `1.2.7`](https://renovatebot.com/diffs/npm/minimist/1.2.7/1.2.7) |
| [mocha](https://mochajs.org/) ([source](https://togithub.com/mochajs/mocha)) | devDependencies | pin | [`^10.2.0` -> `10.2.0`](https://renovatebot.com/diffs/npm/mocha/10.2.0/10.2.0) |
| [mock-fs](https://togithub.com/tschaub/mock-fs) | devDependencies | pin | [`^5.2.0` -> `5.2.0`](https://renovatebot.com/diffs/npm/mock-fs/5.2.0/5.2.0) |
| [nanoid](https://togithub.com/ai/nanoid) | resolutions | pin | [`^3.1.31` -> `3.3.4`](https://renovatebot.com/diffs/npm/nanoid/3.3.4/3.3.4) |
| [node-fetch](https://togithub.com/node-fetch/node-fetch) | resolutions | pin | [`^3.2.10` -> `3.3.0`](https://renovatebot.com/diffs/npm/node-fetch/3.3.0/3.3.0) |
| [node-fetch](https://togithub.com/node-fetch/node-fetch) | dependencies | pin | [`^2.6.7` -> `2.6.7`](https://renovatebot.com/diffs/npm/node-fetch/2.6.7/2.6.7) |
| [open](https://togithub.com/sindresorhus/open) | dependencies | pin | [`^8.4.0` -> `8.4.0`](https://renovatebot.com/diffs/npm/open/8.4.0/8.4.0) |
| [parse-path](https://togithub.com/IonicaBizau/parse-path) | resolutions | pin | [`^5.0.0` -> `5.0.0`](https://renovatebot.com/diffs/npm/parse-path/5.0.0/5.0.0) |
| [parse-url](https://togithub.com/IonicaBizau/parse-url) | resolutions | pin | [`^8.1.0` -> `8.1.0`](https://renovatebot.com/diffs/npm/parse-url/8.1.0/8.1.0) |
| [pascalcase](https://togithub.com/jonschlinkert/pascalcase) | dependencies | pin | [`^1.0.0` -> `1.0.0`](https://renovatebot.com/diffs/npm/pascalcase/1.0.0/1.0.0) |
| [proxyquire](https://togithub.com/thlorenz/proxyquire) | devDependencies | pin | [`^2.1.3` -> `2.1.3`](https://renovatebot.com/diffs/npm/proxyquire/2.1.3/2.1.3) |
| [qs](https://togithub.com/ljharb/qs) | resolutions | pin | [`^6.5.3` -> `6.11.0`](https://renovatebot.com/diffs/npm/qs/6.11.0/6.11.0) |
| [qs](https://togithub.com/ljharb/qs) | dependencies | pin | [`^6.11.0` -> `6.11.0`](https://renovatebot.com/diffs/npm/qs/6.11.0/6.11.0) |
| [release-it](https://togithub.com/release-it/release-it) | devDependencies | pin | [`^15.6.0` -> `15.6.0`](https://renovatebot.com/diffs/npm/release-it/15.6.0/15.6.0) |
| [remark-cli](https://remark.js.org) ([source](https://togithub.com/remarkjs/remark)) | devDependencies | pin | [`^11.0.0` -> `11.0.0`](https://renovatebot.com/diffs/npm/remark-cli/11.0.0/11.0.0) |
| [remark-frontmatter](https://togithub.com/remarkjs/remark-frontmatter) | devDependencies | pin | [`^4.0.1` -> `4.0.1`](https://renovatebot.com/diffs/npm/remark-frontmatter/4.0.1/4.0.1) |
| [remark-lint-no-undefined-references](https://togithub.com/remarkjs/remark-lint) | devDependencies | pin | [`^4.2.0` -> `4.2.0`](https://renovatebot.com/diffs/npm/remark-lint-no-undefined-references/4.2.0/4.2.0) |
| [request-promise-native](https://togithub.com/request/request-promise-native) | dependencies | pin | [`^1.0.9` -> `1.0.9`](https://renovatebot.com/diffs/npm/request-promise-native/1.0.9/1.0.9) |
| [semver](https://togithub.com/npm/node-semver) | dependencies | pin | [`^7.3.8` -> `7.3.8`](https://renovatebot.com/diffs/npm/semver/7.3.8/7.3.8) |
| [sha.js](https://togithub.com/crypto-browserify/sha.js) | dependencies | pin | [`^2.4.11` -> `2.4.11`](https://renovatebot.com/diffs/npm/sha.js/2.4.11/2.4.11) |
| [sharp-cli](https://togithub.com/vseventer/sharp-cli) | devDependencies | pin | [`^4.1.0` -> `4.1.0`](https://renovatebot.com/diffs/npm/sharp-cli/4.1.0/4.1.0) |
| [source-map](https://togithub.com/mozilla/source-map) | dependencies | pin | [`^0.8.0-beta.0` -> `0.8.0-beta.0`](https://renovatebot.com/diffs/npm/source-map/0.8.0-beta.0/0.8.0-beta.0) |
| [stacktrace-parser](https://togithub.com/errwischt/stacktrace-parser) | dependencies | pin | [`^0.1.10` -> `0.1.10`](https://renovatebot.com/diffs/npm/stacktrace-parser/0.1.10/0.1.10) |
| [ts-loader](https://togithub.com/TypeStrong/ts-loader) | devDependencies | pin | [`^9.4.2` -> `9.4.2`](https://renovatebot.com/diffs/npm/ts-loader/9.4.2/9.4.2) |
| [ts-node](https://typestrong.org/ts-node) ([source](https://togithub.com/TypeStrong/ts-node)) | devDependencies | pin | [`^10.9.1` -> `10.9.1`](https://renovatebot.com/diffs/npm/ts-node/10.9.1/10.9.1) |
| [typedoc](https://typedoc.org) ([source](https://togithub.com/TypeStrong/TypeDoc)) | devDependencies | pin | [`^0.23.10` -> `0.23.10`](https://renovatebot.com/diffs/npm/typedoc/0.23.10/0.23.10) |
| [typedoc-plugin-markdown](https://togithub.com/tgreyuk/typedoc-plugin-markdown/tree/master/packages/typedoc-plugin-markdown) ([source](https://togithub.com/tgreyuk/typedoc-plugin-markdown)) | devDependencies | pin | [`^3.14.0` -> `3.14.0`](https://renovatebot.com/diffs/npm/typedoc-plugin-markdown/3.14.0/3.14.0) |
| [typescript](https://www.typescriptlang.org/) ([source](https://togithub.com/Microsoft/TypeScript)) | dependencies | pin | [`^4.7.4` -> `4.7.4`](https://renovatebot.com/diffs/npm/typescript/4.7.4/4.7.4) |
| [uglify-js](https://togithub.com/mishoo/UglifyJS) | dependencies | pin | [`^3.17.4` -> `3.17.4`](https://renovatebot.com/diffs/npm/uglify-js/3.17.4/3.17.4) |
| [url-parse](https://togithub.com/unshiftio/url-parse) | dependencies | pin | [`^1.5.10` -> `1.5.10`](https://renovatebot.com/diffs/npm/url-parse/1.5.10/1.5.10) |
| [uuid](https://togithub.com/uuidjs/uuid) | dependencies | pin | [`^9.0.0` -> `9.0.0`](https://renovatebot.com/diffs/npm/uuid/9.0.0/9.0.0) |
| [xml2js](https://togithub.com/Leonidas-from-XIV/node-xml2js) | dependencies | pin | [`^0.4.23` -> `0.4.23`](https://renovatebot.com/diffs/npm/xml2js/0.4.23/0.4.23) |
| [yargs](https://yargs.js.org/) ([source](https://togithub.com/yargs/yargs)) | dependencies | pin | [`^17.7.1` -> `17.7.1`](https://renovatebot.com/diffs/npm/yargs/17.7.1/17.7.1) |
| [yn](https://togithub.com/sindresorhus/yn) | dependencies | pin | [`^3.1.1` -> `3.1.1`](https://renovatebot.com/diffs/npm/yn/3.1.1/3.1.1) |

Add the preset `:preserveSemverRanges` to your config if you don't want to pin your dependencies.

---

### Configuration

📅 **Schedule**: Branch creation - "before 3am on the first day of the month" (UTC), Automerge - At any time (no schedule defined).

🚦 **Automerge**: Enabled.

♻ **Rebasing**: Whenever PR is behind base branch, or you tick the rebase/retry checkbox.

👻 **Immortal**: This PR will be recreated if closed unmerged. Get [config help](https://togithub.com/renovatebot/renovate/discussions) if that's undesired.

---

 - [ ] If you want to rebase/retry this PR, check this box

---

This PR has been generated by [Self-Hosted Renovate Bot](https://togithub.com/coda/renovate-config). Find all logs [here](https://app.circleci.com/pipelines/github/coda/renovate-config?branch=main).

### CODE REVIEW

**Comment by krypton-build-user** [comment]
### ⚠ Artifact update problem

Renovate failed to update an artifact related to this branch. You probably do not want to merge this PR as-is.

♻ Renovate will retry this branch, including artifacts, only when one of the following happens:

 - any of the package files in this branch needs updating, or 
 - the branch becomes conflicted, or
 - you click the rebase/retry checkbox if found above, or
 - you rename this PR's title to start with "rebase!" to trigger it manually

The artifact failure details are included below:

##### File name: yarn.lock

```
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?

```

**Comment by krypton-build-user** [comment]
### ⚠ Artifact update problem

Renovate failed to update an artifact related to this branch. You probably do not want to merge this PR as-is.

♻ Renovate will retry this branch, including artifacts, only when one of the following happens:

 - any of the package files in this branch needs updating, or 
 - the branch becomes conflicted, or
 - you click the rebase/retry checkbox if found above, or
 - you rename this PR's title to start with "rebase!" to trigger it manually

The artifact failure details are included below:

##### File name: yarn.lock

```
You don't have permission to create /tmp/renovate/cache/containerbase.
curl: (23) Failed writing received data to disk/application
Download failed: https://nodejs.org/dist/v18.14.2/SHASUMS256.txt

```

File: package.json [hunk 1]
```diff
@@ -18,117 +18,117 @@
...
+    "url-parse": "1.5.10",
+    "uuid": "9.0.0",
+    "xml2js": "0.4.23",
+    "yargs": "17.7.1",
+    "yn": "3.1.1",
     "zod": "3.20.2"
   },
   "optionalDependencies": {
-    "isolated-vm": "=4.4.1 || >4.4.2"
+    "isolated-vm": "4.4.1"
```
Comment (dweitzman-codaio) [inline package.json:57]: We can't put a comment into package.json to remind people not to upgrade to 4.4.2. 4.4.3 isn't released yet, so maybe we just have to collectively remember until 4.4.3 is out

File: yarn.lock [hunk 2]
```diff
@@ -2659,16 +2659,16 @@ cliui@^8.0.1:
     strip-ansi "^6.0.1"
     wrap-ansi "^7.0.0"
 
+clone@2.1.2:
+  version "2.1.2"
+  resolved "https://registry.yarnpkg.com/clone/-/clone-2.1.2.tgz#1b7f4b9f591f1e8f83670401600345a02887435f"
+  integrity sha512-3Pe/CF1Nn94hyhIYpjtiLhdCoEoz0DqQ+988E9gmeEdQZlojxnOb74wctFyuwWQHzqyf9X7C7MG8juUpqBJT8w==
```
Comment (dweitzman-codaio) [inline yarn.lock:2665]: For clone@2.1.2 the checksum here changed from sha1 to sha512, so I verified manually that downloading both the npmjs.org link and the yarnpkg.com link returns the exact same file.

File: yarn.lock [hunk 3]
```diff
@@ -5421,10 +5421,10 @@ merge2@^1.3.0, merge2@^1.4.1:
   resolved "https://registry.npmjs.org/merge2/-/merge2-1.4.1.tgz"
   integrity sha512-8q7VEgMJW4J8tcfVPy8g09NcQwZdbwFEqhe/WZkoIzjn/3TGDwtOCYtXGxA3O8tPzpczCCDgv+P2P5y00ZJOOg==
 
-mersenne-twister@^1.1.0:
+mersenne-twister@1.1.0:
   version "1.1.0"
-  resolved "https://registry.npmjs.org/mersenne-twister/-/mersenne-twister-1.1.0.tgz"
-  integrity sha1-+RZhjuQ9cXnvz2Qb7EUx65Zwl4o=
+  resolved "https://registry.yarnpkg.com/mersenne-twister/-/mersenne-twister-1.1.0.tgz#f916618ee43d7179efcf641bec4531eb9670978a"
+  integrity sha512-mUYWsMKNrm4lfygPkL3OfGzOPTR2DBlTkBNHM//F6hGp8cLThY897crAlk3/Jo17LEOOjQUrNAx6DvgO77QJkA==
```
Comment (dweitzman-codaio) [inline yarn.lock:5427]: Same with mersenne-twister. I manually verified that the npmjs and yarnpack files are identical even though renovate switched from sha1 to sha256

**Review by dweitzman-codaio — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (jonathan-codaio) [reply package.json:57]: Hmm @chris-codaio does Renovate really not have a way to indicate a bad version that you want to ignore? I'd rather not rely on us having to remember this, the bug here was very subtle and I think only affected external developers on another platform and hard to prepare a test for.

Comment (dweitzman-codaio) [reply package.json:57]: Ah, yes we can do it in renovate.json

Comment (dweitzman-codaio) [reply package.json:57]: https://github.com/coda/packs-sdk/pull/2480

**Pull request merged**